src/object/cli_obj.c

```diff
@@ -4504,17 +4510,33 @@ obj_list_get_shard(struct obj_auxi_args *obj_auxi, unsigned int map_ver,
 			DP_OID(obj->cob_md.omd_id), shard);
 
 		obj_auxi->spec_shard = 1;
+		obj_auxi->spec_group = 0;
+
 		D_GOTO(out, shard);
 	}
 
 	if (args->dkey_anchor != NULL &&
-	    (daos_anchor_get_flags(args->dkey_anchor) & DIOF_TO_SPEC_SHARD)) {
+	    daos_anchor_get_flags(args->dkey_anchor) & DIOF_TO_SPEC_SHARD) {
 		shard = dc_obj_anchor2shard(args->dkey_anchor);
+		obj_auxi->specified_shard = shard;
 		obj_auxi->spec_shard = 1;
+		obj_auxi->spec_group = 0;
+
+		D_GOTO(out, shard);
+	}
+
+	if (args->dkey_anchor != NULL &&
+	    daos_anchor_get_flags(args->dkey_anchor) & DIOF_TO_SPEC_GROUP) {
+		shard = dc_obj_anchor2shard(args->dkey_anchor);
+		obj_auxi->spec_shard = 0;
+		obj_auxi->spec_group = 1;
 		grp_idx = shard / obj_get_replicas(obj);
 	} else {
 		uint64_t dkey_hash;
 
+		obj_auxi->spec_shard = 0;
+		obj_auxi->spec_group = 0;
```

Comment:
I thought obj_auxi is initialized by 0.

Reply:
The push/pop does not guarantee the initialization.